source/dub/packagesuppliers/filesystem.d

```diff
@@ -27,7 +27,7 @@ class FileSystemPackageSupplier : PackageSupplier {
 		import std.file : dirEntries, DirEntry, SpanMode;
 		import std.conv : to;
 		Version[] ret;
-		foreach (DirEntry d; dirEntries(m_path.toNativeString(), package_id~"*", SpanMode.shallow)) {
+		foreach (DirEntry d; dirEntries(m_path.toNativeString(), package_id~"-[0123456789]*", SpanMode.shallow)) {
```

Comment:
I appreciate the code golf aspect of this, but we should probably rather check for `isValidVersion(vers)` below instead (which comes from `dub.semver`)

Otherwise something like [open-simplex-2](https://code.dlang.org/packages/open-simplex-2) would wrongly match with `open-simplex` (currently unregistered) as package name

Reply:
I get you. I was under the impression that dub doesn't allow digits in the name of the package